managed/resource_api_key.go

```diff
@@ -0,0 +1,318 @@
+	diags.Append(plan.GetAttribute(ctx, path.Root("role_name"), &apiKey.RoleName)...)
+	diags.Append(plan.GetAttribute(ctx, path.Root("description"), &apiKey.Description)...)
+
+	return diags
+}
+
+func GetKeyTimeConversionMap() map[string]int {
+	return map[string]int{
+		"Hours":  1,
+		"Days":   24,
+		"Months": 24 * 30,
+	}
+}
+
+// Create API Key
+func (r resourceApiKey) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var plan ApiKey
+	var accountId, message string
+	var getAccountOK bool
+	resp.Diagnostics.Append(getApiKeyPlan(ctx, req.Plan, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the API Key")
+		return
+	}
+
+	if (!plan.ApiKeyID.Unknown && !plan.ApiKeyID.Null) || plan.ApiKeyID.Value != "" {
+		resp.Diagnostics.AddError(
+			"API Key ID provided for new API Key",
+			"The api_key_id was provided even though a new API Key is being issued. Do not include this field in the provider when creating an API Key.",
+		)
+		return
+	}
+
+	apiClient := r.p.client
+
+	accountId, getAccountOK, message = getAccountId(ctx, apiClient)
+	if !getAccountOK {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, getProjectOK, message := getProjectId(ctx, apiClient, accountId)
+	if !getProjectOK {
+		resp.Diagnostics.AddError("Unable to get project ID ", message)
+		return
+	}
+
+	name := plan.Name.Value
+
+	duration := plan.Duration.Value
+	unit := plan.Unit.Value
+	var expiryHours int
+	validTimeUnit := false
+	for k, v := range GetKeyTimeConversionMap() {
+		if unit == k {
+			validTimeUnit = true
+			expiryHours = int(duration) * v
+		}
+	}
+	if !validTimeUnit {
+		resp.Diagnostics.AddError(
+			"Invalid time unit for API Key creation",
+			"Available options are Hours, Days, and Months.",
+		)
+	}
+
+	roleName := plan.RoleName.Value
+
+	roleResp, response, err := apiClient.RoleApi.ListRbacRoles(ctx, accountId).RoleTypes("ALL").Limit(100).DisplayName(roleName).Execute()
+	if err != nil {
+		errMsg := getErrorMessage(response, err)
+		resp.Diagnostics.AddError("Unable to create API Key", errMsg)
+		return
+	}
+	roleId := roleResp.Data[0].Info.Id
+
+	apiKeySpec := openapiclient.NewApiKeySpec(name, int32(expiryHours))
+	apiKeySpec.SetRoleId(roleId)
+
+	if plan.Description.Value != "" {
+		apiKeySpec.SetDescription(plan.Description.Value)
+	}
+
+	apiKeyResp, response, err := apiClient.AuthApi.CreateApiKey(ctx, accountId).ApiKeySpec(*apiKeySpec).Execute()
+	if err != nil {
+		errMsg := getErrorMessage(response, err)
+		resp.Diagnostics.AddError("Unable to create API Key ", errMsg)
+		return
+	}
+	apiKeyId := apiKeyResp.Data.Info.Id
+
```

Comment:
What is the best way to show the issued API Key to the TF users only once without persisting it in the state?
Something like https://github.com/yugabyte/ybm-cli/blob/64fe2dcc7e4d43e1c81b1cc86cc053edfca9d813/cmd/api_key/api_key.go#L180

Reply:
It's one of the terraform limitations, same with AWS Key and so on.. you put in. the state or not, but showing in the output is even worse..

Reply:
Updated to keep the key in state